sbt-plugin/src/main/scala/akka/grpc/sbt/AkkaGrpcPlugin.scala

```diff
@@ -69,9 +81,52 @@ object AkkaGrpcPlugin extends AutoPlugin {
           unmanagedResources := { Defaults.collectFiles(unmanagedResourceDirectories, includeFilter, excludeFilter).value },
           resources := managedResources.value ++ unmanagedResources.value)))
 
-  def adaptAkkaGenerator(targetPath: File)(generatorAndSettings: GeneratorAndSettings): Target = {
-    val adapted = new ProtocBridgeSbtPluginCodeGenerator(generatorAndSettings.generator)
-    val generator = JvmGenerator(generatorAndSettings.generator.name, adapted)
-    (generator, generatorAndSettings.settings) -> targetPath
+  private def targetsFor(targetPath: File, settings: Seq[String], stubs: Seq[AkkaGrpc.GeneratedSource], languages: Seq[AkkaGrpc.Language]): Seq[protocbridge.Target] = {
+    val generators = generatorsFor(stubs, languages, settings)
+    generators.map { case (generator, generatorSettings) => protocbridge.Target(generator, targetPath, generatorSettings) }
+  }
+
+  // creates a seq of generator and per generator settings
+  private def generatorsFor(stubs: Seq[AkkaGrpc.GeneratedSource], languages: Seq[AkkaGrpc.Language], defaultSettings: Seq[String]): Seq[(protocbridge.Generator, Seq[String])] = {
+    // these two are the model/message (protoc) generators
+    def ScalaGenerator: (protocbridge.Generator, Seq[String]) = (protocbridge.JvmGenerator("scala", ScalaPbCodeGenerator), defaultSettings)
+    // we have a default flat_package, but that doesn't play with the java generator (it fails)
+    def JavaGenerator: (protocbridge.Generator, Seq[String]) = (PB.gens.java, defaultSettings.filterNot(_ == "flat_package"))
```

Comment:
If the Java generator crashes on parameters it doesn't understand I think we should whitelist only the parameters we know it understands, rather than blacklisting the ones we know of that it doesn't understand. We do something similar for the scalapb plugin in `ScalaCodeGenerator`: that also crashes on parameters it doesn't expect.

OK for now though, it's not the point of this PR after all ;).